enterprise/api_client/enterprise_catalog.py

```diff
@@ -110,6 +113,60 @@ def get_enterprise_catalog(self, catalog_uuid, should_raise_exception=True):
                 )
             return {}
 
+    @UserAPIClient.refresh_token
+    def get_enterprise_catalog_by_hash(self, catalog_query_hash, should_raise_exception=True):
+        """
+        Gets an enterprise catalog by its content query hash.
+
+        Arguments:
+            catalog_query_hash (str): The hash of an EnterpriseCatalog instance's query
+            should_raise_exception (bool): Whether an exception should be logged if
+                a catalog does not yet exist in enterprise-catalog. Defaults to True.
+
+        Returns:
+            dict: a dictionary representing an enterprise catalog
+        """
+        api_url = self.get_api_url(self.GET_QUERY_BY_HASH_ENDPOINT.format(catalog_query_hash), append_slash=False)
+        try:
+            response = self.client.get(api_url)
+            response.raise_for_status()
+            return response.json()
+        except (RequestException, ConnectionError, Timeout) as exc:
+            LOGGER.exception(
+                'Failed to get EnterpriseCustomer Catalog by hash [%s] in enterprise-catalog due to: [%s]',
+                catalog_query_hash, str(exc)
+            )
+            if should_raise_exception:
+                raise
+            return {}
+
+    @UserAPIClient.refresh_token
+    def get_catalog_query_hash(self, catalog_query, should_raise_exception=True):
+        """
+        Gets hash for content query.
+
+        Arguments:
+            catalog_query (obj): content query json
+            should_raise_exception (bool): Whether an exception should be logged if
+                a catalog does not yet exist in enterprise-catalog. Defaults to True.
+
+        Returns:
+            dict: a dictionary representing an enterprise catalog
+        """
+        api_url = self.get_api_url(self.GET_CONTENT_FILTER_HASH_ENDPOINT, append_slash=False)
+        try:
+            response = self.client.get(api_url, data=catalog_query)
+            response.raise_for_status()
+            return response.json()
+        except (RequestException, ConnectionError, Timeout) as exc:
+            LOGGER.exception(
+                'Failed to get catalog query hash for "[%s]" due to: [%s]',
+                catalog_query, str(exc)
+            )
+            if should_raise_exception:
```

Comment:
Will this end up raising an exception if the call returns a 404 and `should_raise_exception==True`?

Reply:
It will raise an exception regardless of `should_raise_exception`, keeping with the behavior of other enterprise-catalog client methods.